### PR TITLE
optimize: put the xid and branchId into the MDC

### PR DESCRIPTION
**Pull request by wangliang181230**
### Ⅰ. Describe what this PR did
optimize: put the xid and branchId into the MDC
优化：将xid和branchId放入MDC中，能够通过xid找到与全局事务相关的所有日志。

### CODE REVIEW

File: core/src/main/java/io/seata/core/context/RootContext.java [hunk 1]
```diff
@@ -90,6 +101,8 @@ public static String getXID() {
     public static void bind(@Nonnull String xid) {
         if (StringUtils.isBlank(xid)) {
             xid = null;
```
[the remark this answers is not in the document]
Comment (wangliang181230) [reply core/src/main/java/io/seata/core/context/RootContext.java:103]: 哦，对，这里兼具着unbind的功能。

[the remark this answers is not in the document]
Comment (wangliang181230) [reply core/src/main/java/io/seata/core/context/RootContext.java]: done

File: server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java [hunk 2]
```diff
@@ -164,30 +167,35 @@ protected void doGlobalBegin(GlobalBeginRequest request, GlobalBeginResponse res
...
         response.setGlobalStatus(core.globalReport(request.getXid(), request.getGlobalStatus()));
     }
 
     @Override
     protected void doBranchRegister(BranchRegisterRequest request, BranchRegisterResponse response,
                                     RpcContext rpcContext) throws TransactionException {
+        MDC.put(RootContext.MDC_KEY_XID, request.getXid());
         response.setBranchId(
             core.branchRegister(request.getBranchType(), request.getResourceId(), rpcContext.getClientId(),
                 request.getXid(), request.getApplicationData(), request.getLockKey()));
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java:201]: How about putting after you get branchId?

Comment (wangliang181230) [reply server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java:201]: After the branch transaction ID is generated, it has been put into MDC in the `AbstractCore.java `

**Comment by slievrly** [comment]
image

**Comment by wangliang181230** [comment]
> 
> 
> image

done

> 
> 
> image

done

File: server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java [hunk 3]
```diff
@@ -242,14 +253,14 @@ protected void timeoutCheck() throws TransactionException {
                 return true;
             });
             if (!shouldTimeout) {
-                continue;
+                // continue
+                return;
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/coordinator/DefaultCoordinator.java:257]: why only check the first globalSession?

File: server/src/main/java/io/seata/server/session/SessionHelper.java [hunk 4]
```diff
@@ -139,4 +146,48 @@ public static boolean isTimeoutGlobalStatus(GlobalStatus status) {
...
+        for (BranchSession branchSession : sessions) {
+            try {
+                MDC.put(RootContext.MDC_KEY_BRANCH_ID, String.valueOf(branchSession.getBranchId()));
+                result = handler.handle(branchSession);
+                if (result == null) {
+                    continue;
+                }
+                return result;
+            } finally {
+                MDC.remove(RootContext.MDC_KEY_XID);
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/session/SessionHelper.java:188]: MDC_KEY_BRANCH_ID

Comment (wangliang181230) [reply server/src/main/java/io/seata/server/session/SessionHelper.java:188]: fixed

**Review by slievrly — APPROVED** [review]
LGTM

**Review by xingfudeshi — APPROVED** [review]
LGTM.

**Pull request merged**